screens/dashboard/components/upcoming.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import moment from 'moment';
+
+// import global actions
+import * as actions from '../actions';
+
+// import global styles
+// @TODO: move the global styles from this screen into ../../styles/common 
+import commonStyles from '../../../styles/common';
+const styles = StyleSheet.create(commonStyles);
+
+type Props = {
+  actions: Object,
+  profile: Object,
+  navigation: Object
+};
+
+
+export class Upcoming extends Component<Props> {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+    this.icons = {
```

Comment:
You don't need to assign these icons to property on the class.   This makes them public and opens the door to another developer changing your icons.  I'm assuming you don't want that.

Assign them to a constant outside the class.  That will make them private, but still available for use inside the class.   Using closures to make things private is a common pattern in JS.

Reply:
If you're unclear about closures and scope I'd recommend  this article by JS Guru  Kyle Simpson:

https://github.com/getify/You-Dont-Know-JS/blob/master/scope%20%26%20closures/ch3.md

Reply:
```
const icons = {
  'open': 'angle-down',
  'close': 'angle-up'
};

export class Upcoming extends Component<Props> {
```